spec/lib/diaspora/federated/generator_spec.rb

```diff
@@ -0,0 +1,17 @@
+require "spec_helper"
+
+describe "comment on reshare" do
+  before do
+    @status_message = FactoryGirl.build(:status_message, author: alice.person, public: true)
+    @reshare = FactoryGirl.build(:reshare, root: @status_message, author: bob.person)
+    @comment = FactoryGirl.build(:comment, post: @reshare, text: "reshare comment")
+  end
+
+  it "delivers comment also to root author" do
+    # opts = {:additional_subscribers => [alice.person]}
+    allow_any_instance_of(Federated::Generator).to receive(:create!)
+    # expect(Postzord::Dispatcher).to receive(:defer_build_and_post).with(bob, @comment, opts)
+    expect(Postzord::Dispatcher).to receive(:defer_build_and_post)
```

Comment:
Thanks @jhass for the review. I still can't figure out this however, how to get `@comment.save!` (which I assume is the right way to go from here) to end up calling `defer_build_and_post`.

I assume I have to do some allow etc magic, but, as always, rspec got the better of me. I'll try reading up on it during the weekend, or if you have a quick hint, appreciated. Maybe the code doesn't work yet, but since the test doesn't even end up in that section (verified via debugger), difficult to say for sure.